### PR TITLE
ENG-429 fix resource group creation for azure

### PR DESCRIPTION
**Pull request by zreigz**
## Summary
Create a resource group if doesn't exist for the Azure provider
Unit tests

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [X] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: pkg/provider/azure_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-06-01/storage"
+	"github.com/pluralsh/plural/pkg/manifest"
+	"github.com/pluralsh/plural/pkg/provider"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateResourceGroup(t *testing.T) {
+	resourceGroupName := "test"
+	tests := []struct {
```
Comment (michaeljguarino) [inline pkg/provider/azure_test.go:15]: could we not define these structs inline (can be separate private structs in this file)?  Will just reduce the line lengths of these tests and improve readability a bit I think

File: pkg/provider/azure.go [hunk 2]
```diff
@@ -3,35 +3,85 @@ package provider
...
 	"github.com/pluralsh/plural/pkg/template"
 	"github.com/pluralsh/plural/pkg/utils"
-	"github.com/pluralsh/plural/pkg/utils/errors"
+	pluralerr "github.com/pluralsh/plural/pkg/utils/errors"
 	v1 "k8s.io/api/core/v1"
 )
 
+// ResourceGroupClient is the subset of functions we need from armresources.VirtualResourceGroupsClient;
+// this interface is purely here for allowing unit tests.
+type ResourceGroupClient interface {
```
[the remark this answers is not in the document]
Comment (zreigz) [reply pkg/provider/azure.go:35]: unfortunately not, azure's `armresources.ResourceGroupsClient` is a struct like the other clients

File: pkg/provider/azure_test.go (hunk 1 above)
Comment (zreigz) [reply pkg/provider/azure_test.go:15]: When we have one test scenario in the file then we can define the struct with the test parameters:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/convert/convert_test.go#L31

 But for multiple test functions, it's a common pattern to use an inline struct for the test parameters: https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/kubelet/kuberuntime/helpers_linux_test.go

File: pkg/provider/azure.go [hunk 3]
```diff
@@ -127,13 +181,26 @@ func mkAzure(conf config.Config) (prov *AzureProvider, err error) {
...
+	var err error
+	clients := clientSet
+	if clientSet == nil {
+		clients, err = GetClientSet(utils.ToString(man.Context["SubscriptionId"]))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &AzureProvider{man.Cluster, man.Project, man.Bucket, man.Region, man.Context, nil, clients}, nil
```
Comment (DavidSpek) [inline pkg/provider/azure.go:194]: I'm not sure if we need to initialize the client here, since we also don't initialize the writer.

Comment (zreigz) [reply pkg/provider/azure.go:194]: we have to initialize the clients here for two reasons:
 - to initialize fake clients for the tests (AzureProvider has private fields and can be done in the same package) and it's the main reason
 - previously the clients were created on demand before the call, now after refactor they must exist when we initialize the Azure provider

Comment (DavidSpek) [reply pkg/provider/azure.go:194]: Now that I think about it I believe I did something similar with the AWS provider a while back.

File: pkg/provider/azure.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (michaeljguarino) [reply pkg/provider/azure.go:35]: Ugh

File: pkg/provider/azure_test.go (hunk 1 above)
Comment (michaeljguarino) [reply pkg/provider/azure_test.go:15]: If it's common golang style then that's cool, just makes the func a bit longer

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**